pages/Product.vue

```diff
@@ -182,7 +181,13 @@ export default {
     registerModule(ReviewModule);
     registerModule(RecentlyViewedModule);
   },
+  beforeMount () {
+    this.banners = this.promotedOffers.productBanners
+  },
   async mounted () {
+    supportsWebP.then(supported => {
```

Comment:
same proposition as above to make one check in some module initialization and then store it in vuex.
and maybe move everything to computed and `createBanners ` to helper?

Reply:
yes